src/ContentHubClient.php

```diff
@@ -340,514 +446,583 @@ public function putEntity(CDFObject $object)
-     *
-     * @throws \GuzzleHttp\Exception\RequestException
-     */
-    public function searchEntity($query)
-    {
-        $options['body'] = json_encode((array) $query);
-        return $this->getResponseJson($this->get('_search', $options));
-    }
+    // Now make the request.
+    return $this->getResponseJson($this->get($url));
+  }
 
-    /**
-     * Returns the Client, given the site name.
-     *
-     * @param string $name
-     *
-     * @return mixed
-     */
-    public function getClientByName($name)
-    {
-        return $this->getResponseJson($this->get("settings/clients/$name"));
-    }
+  /**
+   * Searches for entities.
+   *
+   * @param array $query
+   *   Search query options.
+   *
+   * @return mixed
+   *   Operation result.
+   *
+   * @throws \GuzzleHttp\Exception\RequestException
+   * @throws \Exception
+   */
+  public function searchEntity($query) {
+    $options['body'] = json_encode((array) $query);
+    return $this->getResponseJson($this->get('_search', $options));
+  }
 
-    public function getClients()
-    {
-      $data = $this->getResponseJson($this->get('settings'));
-      return $data['clients'];
-    }
+  /**
+   * Returns the Client, given the site name.
+   *
+   * @param string $name
+   *   Client name.
+   *
+   * @return mixed
+   *   Client info.
+   *
+   * @throws \Exception
+   */
+  public function getClientByName($name) {
+    return $this->getResponseJson($this->get("settings/clients/$name"));
+  }
 
-    public function getWebHooks()
-    {
-        $data = $this->getResponseJson($this->get('settings'));
-        return $data['webhooks'];
-    }
+  /**
+   * Reruns clients list.
+   *
+   * @return mixed
+   *   Clients list.
+   *
+   * @throws \Exception
+   */
+  public function getClients() {
+    $data = $this->getResponseJson($this->get('settings'));
+    return $data['clients'];
+  }
 
-    public function getWebHook($url)
-    {
-        $webhooks = $this->getWebHooks();
-        foreach ($webhooks as $webhook) {
-            if ($webhook['url'] == $url) {
-                return $webhook;
-            }
-        }
-        return [];
-    }
+  /**
+   * Returns webhooks.
+   *
+   * @return mixed
+   *   Webhooks info.
+   *
+   * @throws \Exception
+   */
+  public function getWebHooks() {
+    $data = $this->getResponseJson($this->get('settings'));
+    return $data['webhooks'];
+  }
 
-    /**
-     * Get the settings that were used to instantiate this client.
-     *
-     * @return \Acquia\ContentHubClient\Settings
-     */
-    public function getSettings() {
-      return $this->settings;
+  /**
+   * Returns webhook by given URL.
+   *
+   * @param string $url
+   *   URL.
+   *
+   * @return array
+   *   Webhook info.
+   *
+   * @throws \Exception
+   */
+  public function getWebHook($url) {
+    $webhooks = $this->getWebHooks();
+    foreach ($webhooks as $webhook) {
+      if ($webhook['url'] == $url) {
+        return $webhook;
+      }
     }
```

Comment:
I think we can do this without foreach just by using array functions.

Reply:
Agree with you but code refactoring is out of scope of this task.